src/main/ts/ephox/photon/Navigation.ts

```diff
@@ -1,21 +1,26 @@
-import { document } from '@ephox/dom-globals';
+import { Document, document, Element as DomElement, Node } from '@ephox/dom-globals';
 import { Option } from '@ephox/katamari';
 import { Element, Traverse } from '@ephox/sugar';
 
-var view = function (doc) {
+export interface Navigation<E, D> {
+  view: (doc: D) => Option<E>;
```

Comment:
Fields can be readonly

Reply:
I need to remember to include that more, thanks for the catch!